2019-04-14T19:56:57.1295924+02:00 0HLM10AAUVM8V:00000001 [INF] Request starting HTTP/1.1 GET http://localhost:44376/api/values   (ca22a1cb)
2019-04-14T19:56:57.1295920+02:00 0HLM10AAUVM90:00000001 [INF] Request starting HTTP/1.1 DEBUG http://localhost:51754/  0 (ca22a1cb)
2019-04-14T19:56:57.2264677+02:00 0HLM10AAUVM90:00000001 [INF] Request finished in 103.345ms 200  (791a596a)
2019-04-14T19:56:57.3877607+02:00 0HLM10AAUVM8V:00000001 [INF] Executing endpoint '"TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)"' (500cc934)
2019-04-14T19:56:57.4313551+02:00 0HLM10AAUVM8V:00000001 [INF] Route matched with "{action = \"Get\", controller = \"Values\"}". Executing action "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" (a44c0341)
2019-04-14T19:56:57.5070650+02:00 0HLM10AAUVM8V:00000001 [INF] Executing action method "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" - Validation state: Valid (dad538d7)
2019-04-14T19:57:07.8463530+02:00 0HLM10AAUVM8V:00000001 [INF] Save User UserDTO { User_Id: 4, Login: "bb", Password: "tt", Email: "F", Flag: "f", DateCreate: 04/14/2019 19:57:00, LockAccount: False, Tasks: null, Comments: null } (8ba6a599)
2019-04-14T20:11:39.1770518+02:00 0HLM10IHREQ87:00000001 [INF] Request starting HTTP/1.1 GET http://localhost:44376/api/values   (ca22a1cb)
2019-04-14T20:11:39.1770518+02:00 0HLM10IHREQ86:00000001 [INF] Request starting HTTP/1.1 DEBUG http://localhost:51754/  0 (ca22a1cb)
2019-04-14T20:11:39.2458607+02:00 0HLM10IHREQ86:00000001 [INF] Request finished in 73.5783ms 200  (791a596a)
2019-04-14T20:11:39.3739914+02:00 0HLM10IHREQ87:00000001 [INF] Executing endpoint '"TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)"' (500cc934)
2019-04-14T20:11:39.4106110+02:00 0HLM10IHREQ87:00000001 [INF] Route matched with "{action = \"Get\", controller = \"Values\"}". Executing action "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" (a44c0341)
2019-04-14T20:11:39.4753757+02:00 0HLM10IHREQ87:00000001 [INF] Executing action method "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" - Validation state: Valid (dad538d7)
2019-04-14T20:11:59.2076889+02:00 0HLM10IHREQ87:00000001 [ERR] Error DbUpdateException { Entries: [], Message: "An error occurred while updating the entries. See the inner exception for details.", Data: [], InnerException: SqlException { Errors: [SqlError { Source: "Core .Net SqlClient Data Provider", Number: 515, State: 2, Class: 16, Server: "DESKTOP-7GLL7LJ\OPSDB", Message: "Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.", Procedure: "", LineNumber: 2 }, SqlError { Source: "Core .Net SqlClient Data Provider", Number: 3621, State: 0, Class: 0, Server: "DESKTOP-7GLL7LJ\OPSDB", Message: "The statement has been terminated.", Procedure: "", LineNumber: 2 }], ClientConnectionId: 009dd46b-cfd9-40bc-bb8f-c98a4c949c03, Class: 16, LineNumber: 2, Number: 515, Procedure: "", Server: "DESKTOP-7GLL7LJ\OPSDB", State: 2, Source: "Core .Net SqlClient Data Provider", ErrorCode: -2146232060, Message: "Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.", Data: [DictionaryEntry { Key: "HelpLink.ProdName", Value: "Microsoft SQL Server" }, DictionaryEntry { Key: "HelpLink.ProdVer", Value: "14.00.1000" }, DictionaryEntry { Key: "HelpLink.EvtSrc", Value: "MSSQLServer" }, DictionaryEntry { Key: "HelpLink.EvtID", Value: "515" }, DictionaryEntry { Key: "HelpLink.BaseHelpUrl", Value: "http://go.microsoft.com/fwlink" }, DictionaryEntry { Key: "HelpLink.LinkId", Value: "20476" }], InnerException: null, TargetSite: System.Data.Common.DbDataReader <ExecuteDbDataReaderAsync>b__122_0(System.Threading.Tasks.Task`1[System.Data.SqlClient.SqlDataReader]), StackTrace: "   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__122_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteAsync(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)", HelpLink: null, HResult: -2146232060 }, TargetSite: Void MoveNext(), StackTrace: "   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(DbContext _, ValueTuple`2 parameters, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IReadOnlyList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at TaskMenagerService.DbAccess.TasksDbContext.SaveChangesAsync() in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\DbAccess\TasksDbContext.cs:line 14
   at TaskMenagerService.UserService.Commands.UserHandler.Handle(UserDTO userDTO, CancellationToken cancellationToken) in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\Services\UserService\Commands\UserHandler.cs:line 26", HelpLink: null, Source: "Microsoft.EntityFrameworkCore.Relational", HResult: -2146233088 } (c9135897)
2019-04-14T20:11:59.4301539+02:00 0HLM10IHREQ87:00000001 [INF] Executed action "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" in 20011.2477ms (afa2e885)
2019-04-14T20:11:59.4361368+02:00 0HLM10IHREQ87:00000001 [INF] Executed endpoint '"TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)"' (99874f2b)
2019-04-14T20:11:59.4605335+02:00 0HLM10IHREQ87:00000001 [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.Data.SqlClient.SqlException: Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__122_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteAsync(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at TaskMenagerService.UserService.Commands.UserHandler.Handle(UserDTO userDTO, CancellationToken cancellationToken) in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\Services\UserService\Commands\UserHandler.cs:line 32
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at TaskMenager.API.Controllers.ValuesController.Get() in D:\Projekty\TaskMenager\TaskMenager\TaskMenager.API\Controllers\ValuesController.cs:line 37
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
2019-04-14T20:11:59.5409971+02:00 0HLM10IHREQ87:00000001 [INF] Request finished in 20366.0219ms 500 text/html; charset=utf-8 (791a596a)
2019-04-14T20:33:39.0267331+02:00 0HLM10UR709O7:00000001 [INF] Request starting HTTP/1.1 GET http://localhost:44376/api/values   (ca22a1cb)
2019-04-14T20:33:39.0267327+02:00 0HLM10UR709O6:00000001 [INF] Request starting HTTP/1.1 DEBUG http://localhost:51754/  0 (ca22a1cb)
2019-04-14T20:33:39.0967836+02:00 0HLM10UR709O6:00000001 [INF] Request finished in 73.1755ms 307  (791a596a)
2019-04-14T20:33:39.1271900+02:00 0HLM10UR709O7:00000001 [INF] Executing endpoint '"TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)"' (500cc934)
2019-04-14T20:33:39.1510707+02:00 0HLM10UR709O7:00000001 [INF] Route matched with "{action = \"Get\", controller = \"Values\"}". Executing action "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" (a44c0341)
2019-04-14T20:33:39.2073137+02:00 0HLM10UR709O7:00000001 [INF] Executing action method "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" - Validation state: Valid (dad538d7)
2019-04-14T20:33:39.7164997+02:00 0HLM10UR709O7:00000001 [ERR] Error DbUpdateException { Entries: [], Message: "An error occurred while updating the entries. See the inner exception for details.", Data: [], InnerException: SqlException { Errors: [SqlError { Source: "Core .Net SqlClient Data Provider", Number: 515, State: 2, Class: 16, Server: "DESKTOP-7GLL7LJ\OPSDB", Message: "Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.", Procedure: "", LineNumber: 2 }, SqlError { Source: "Core .Net SqlClient Data Provider", Number: 3621, State: 0, Class: 0, Server: "DESKTOP-7GLL7LJ\OPSDB", Message: "The statement has been terminated.", Procedure: "", LineNumber: 2 }], ClientConnectionId: 09f573f0-df76-40b9-845d-e57ede3c41bb, Class: 16, LineNumber: 2, Number: 515, Procedure: "", Server: "DESKTOP-7GLL7LJ\OPSDB", State: 2, Source: "Core .Net SqlClient Data Provider", ErrorCode: -2146232060, Message: "Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.", Data: [DictionaryEntry { Key: "HelpLink.ProdName", Value: "Microsoft SQL Server" }, DictionaryEntry { Key: "HelpLink.ProdVer", Value: "14.00.1000" }, DictionaryEntry { Key: "HelpLink.EvtSrc", Value: "MSSQLServer" }, DictionaryEntry { Key: "HelpLink.EvtID", Value: "515" }, DictionaryEntry { Key: "HelpLink.BaseHelpUrl", Value: "http://go.microsoft.com/fwlink" }, DictionaryEntry { Key: "HelpLink.LinkId", Value: "20476" }], InnerException: null, TargetSite: System.Data.Common.DbDataReader <ExecuteDbDataReaderAsync>b__122_0(System.Threading.Tasks.Task`1[System.Data.SqlClient.SqlDataReader]), StackTrace: "   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__122_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteAsync(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)", HelpLink: null, HResult: -2146232060 }, TargetSite: Void MoveNext(), StackTrace: "   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(DbContext _, ValueTuple`2 parameters, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IReadOnlyList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at TaskMenagerService.DbAccess.TasksDbContext.SaveChangesAsync() in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\DbAccess\TasksDbContext.cs:line 14
   at TaskMenagerService.UserService.Commands.UserHandler.Handle(UserDTO userDTO, CancellationToken cancellationToken) in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\Services\UserService\Commands\UserHandler.cs:line 25", HelpLink: null, Source: "Microsoft.EntityFrameworkCore.Relational", HResult: -2146233088 } (c9135897)
2019-04-14T20:33:39.7364974+02:00 0HLM10UR709O7:00000001 [INF] Executed action "TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)" in 580.4993ms (afa2e885)
2019-04-14T20:33:39.7391027+02:00 0HLM10UR709O7:00000001 [INF] Executed endpoint '"TaskMenager.API.Controllers.ValuesController.Get (TaskMenager.API)"' (99874f2b)
2019-04-14T20:33:39.7575162+02:00 0HLM10UR709O7:00000001 [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.Data.SqlClient.SqlException: Cannot insert the value NULL into column 'Email', table 'TasksERP.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__122_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteAsync(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at TaskMenagerService.UserService.Commands.UserHandler.Handle(UserDTO userDTO, CancellationToken cancellationToken) in D:\Projekty\TaskMenager\TaskMenager\TaskMenagerService\Services\UserService\Commands\UserHandler.cs:line 31
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at TaskMenager.API.Controllers.ValuesController.Get() in D:\Projekty\TaskMenager\TaskMenager\TaskMenager.API\Controllers\ValuesController.cs:line 37
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
2019-04-14T20:33:39.8052916+02:00 0HLM10UR709O7:00000001 [INF] Request finished in 782.1261ms 500 text/html; charset=utf-8 (791a596a)
